app/controllers/CyaMaintainProfileController.scala

```diff
@@ -0,0 +1,81 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import connectors.TraderProfileConnector
+import controllers.actions._
+import models.{NormalMode, TraderProfile}
+import navigation.Navigator
+import pages.CyaMaintainProfilePage
+import play.api.i18n.MessagesApi
+import play.api.mvc.{Action, AnyContent, Call, MessagesControllerComponents}
+import services.AuditService
+import viewmodels.checkAnswers.HasNirmsSummary
+import viewmodels.govuk.summarylist._
+import views.html.CyaMaintainProfileView
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class CyaMaintainProfileController @Inject() (
+  override val messagesApi: MessagesApi,
+  identify: IdentifierAction,
+  getData: DataRetrievalAction,
+  requireData: DataRequiredAction,
+  val controllerComponents: MessagesControllerComponents,
+  view: CyaMaintainProfileView,
+  traderProfileConnector: TraderProfileConnector,
+  navigator: Navigator,
+  auditService: AuditService
+)(implicit ec: ExecutionContext)
+    extends BaseController {
+
+  private val errorMessage: String = "Unable to update Trader profile."
+  private val continueUrl: Call    = routes.ProfileController.onPageLoad()
+
+  def onPageLoadNirms(): Action[AnyContent] = (identify andThen getData andThen requireData) { implicit request =>
+    TraderProfile.validateHasNirms(request.userAnswers) match {
+      case Right(_)     =>
+        val list = SummaryListViewModel(
+          rows = Seq(
+            HasNirmsSummary.rowUpdate(request.userAnswers)
+          ).flatten
+        )
+        Ok(view(list, routes.CyaMaintainProfileController.onSubmitNirms))
+      case Left(errors) =>
+        logErrorsAndContinue(errorMessage, continueUrl, errors)
+    }
+  }
+
+  def onSubmitNirms(): Action[AnyContent] = (identify andThen getData andThen requireData).async { implicit request =>
+    TraderProfile.validateHasNirms(request.userAnswers) match {
+      case Right(_)     =>
+        traderProfileConnector.getTraderProfile(request.eori).flatMap { traderProfile =>
+          val updatedProfile = traderProfile.copy(nirmsNumber = None)
+          auditService.auditMaintainProfile(traderProfile, updatedProfile, request.affinityGroup)
+          for {
+            _ <- traderProfileConnector.submitTraderProfile(updatedProfile, request.eori)
+          } yield Redirect(navigator.nextPage(CyaMaintainProfilePage, NormalMode, request.userAnswers))
```

Comment:
Do we need to clear the data here?

Reply:
In the ProfileController, the onPageLoad method triggers cleanseProfileData :D